cmd/controller/controllerspc.go

```diff
@@ -0,0 +1,394 @@
+	if err != nil {
+		// The spc resource may no longer exist, in which case we stop
+		// processing.
+		if errors.IsNotFound(err) {
+			runtime.HandleError(fmt.Errorf("spcUpdated '%s' in work queue no longer exists", key))
+			return nil
+		}
+
+		return err
+	}
+
+	var sp apis.StoragePool
+	flag := IsDiskAvailable(spcUpdated.Spec.Name)
+	if flag == false {
+		runtime.HandleError(fmt.Errorf("Disk not available: %s", spcUpdated.Spec.Name))
+		return nil
+	}
+
+	switch operation {
+	case "add":
+		sp = DiskOperations(spcUpdated, namespace)
+		Createsp(sp, c, namespace)
+		break
+	case "update":
+		//IsDiskBeingUsed needs to be implemeneted
+		//IsDiskUnusedMounted needs to be implemeneted
+		sp = DiskOperations(spcUpdated, namespace)
+		Updatesp(sp, c, namespace)
+		break
+	case "delete":
+		//IsDiskBeingUsed needs to be implemeneted
+		//IsDiskUnusedMounted needs to be implemeneted
+		err := block.UnMount(spcUpdated.Spec.Name)
+		if err != nil {
+			runtime.HandleError(fmt.Errorf("Unable to unmount ", err))
+			break
+		}
+		Deletesp(spcUpdated.Spec.Name, c, namespace)
+		break
+	}
+
+	return nil
+}
+
+// enqueueSpc takes a spc resource and converts it into a namespace/name
+// string which is then put onto the work queue. This method should *not* be
+// passed resources of any type other than spc.
+func (c *Controller) enqueueSpc(obj interface{}, q QueueLoad) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	q.key = key
+	c.workqueue.AddRateLimited(q)
+}
+
+func GetNodeName() (string, error) {
+	var Nodename string
+	NodenameByte, err := exec.Command("uname", "-n").Output()
+	Nodename = string(NodenameByte)
+	return Nodename, err
+}
+
+func DiskOperations(spc *apis.StoragePoolClaim, namespace string) apis.StoragePool {
+	Message := ""
+
+	Nodename, err := GetNodeName()
+	if err != nil {
+		Message = Message + "unable to get nodename " + err.Error()
+		//util.CheckErr(err, util.Fatal)
+	}
+
+	res, err := block.Format(spc.Spec.Name, spc.Spec.Format)
+	if err != nil {
+		Message = Message + " unable to format" + err.Error()
+		//util.CheckErr(err, util.Fatal)
+	} else {
+		Message = Message + res
+	}
+
+	mountpoint, err := block.Mount(spc.Spec.Name)
+	if err != nil {
+		Message = Message + " unable to mount" + err.Error()
+		//util.CheckErr(err, util.Fatal)
+	} else {
+		Message = " Mountpoint=" + mountpoint
+		//util.CheckErr(err, util.Fatal)
+	}
+
+	sps := apis.StoragePoolSpec{Name: spc.Spec.Name, Format: spc.Spec.Format,
+		Mountpoint: spc.Spec.Mountpoint,
+		Nodename:   Nodename,
+		Message:    Message,
+	}
+	sp := apis.StoragePool{Spec: sps}
+	sp.Name = sps.Name
+
+	return sp
+}
+
+func Createsp(sp apis.StoragePool, c *Controller, namespace string) {
+	spCopy := sp.DeepCopy()
+	spr, err := c.clientset.NdmV1alpha1().StoragePools().Create(spCopy)
+
+	if err != nil {
+		glog.Info("Unable to create sp", err)
```

Comment:
Will this be an INFO?
Why can't the err be thrown?
How about ERROR & sending an EVENT?